File: pong-game/pong-game/src/paddle.py
import pygame

class Paddle:
    def __init__(self, x, y, width, height, color, dy):
        self.x = x
        self.y = y
        self.width = width
        self.height = height
        self.color = color
        self.dy = dy

    def draw(self, screen):
        pygame.draw.rect(screen, self.color, pygame.Rect(self.x, self.y, self.width, self.height))

    def update(self, height):
        self.y += self.dy

        if self.y < 0:
            self.y = 0
        elif self.y + self.height > height:
            self.y = height - self.height

File: pong-game/src/paddle.js
export class Paddle {
  constructor(x, y, width, height, color) {
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
    this.color = color;
    this.dy = 0;
  }

  draw(ctx) {
    ctx.fillStyle = this.color;
    ctx.fillRect(this.x, this.y, this.width, this.height);
  }

  update(canvasHeight) {
    this.y += this.dy;

    // Keep paddle within canvas bounds
    if (this.y < 0) {
      this.y = 0;
    } else if (this.y + this.height > canvasHeight) {
      this.y = canvasHeight - this.height;
    }
  }

  moveUp() {
    this.dy = -5;
  }

  moveDown() {
    this.dy = 5;
  }

  stop() {
    this.dy = 0;
  }
}